telethon/events/__init__.py

```diff
@@ -236,11 +236,13 @@ def __init__(self, incoming=None, outgoing=None,
         self.incoming = incoming
         self.outgoing = outgoing
         if isinstance(pattern, str):
-            self.pattern = re.compile(pattern).match
-        elif not pattern or callable(pattern):
+            self.pattern = [re.compile(pattern).match]
+        elif isinstance(pattern, (list, tuple)):
             self.pattern = pattern
+        elif not pattern or callable(pattern):
+            self.pattern = [pattern]
```

Comment:
`if not pattern` may succeed with `None`. Then you have `self.pattern = [None]`. This evaluates to `True`, but doing `pattern(x)` will fail.